lib/awsv4/header.js

```diff
@@ -19,33 +19,42 @@ function fillOptions(options) {
 function signWithHeaders(method, service, region = "", target = "", path = "", body = null, query = "", headers = {}) {
   var options = { headers: headers }
   options = fillOptions(options);
+  options.headers["X-Amz-Date"] = toTime(options.timestamp)
+  var host = service.concat(".", options.region, ".amazonaws.com")
+  options.headers["Host"] = host
+
+  if (options.sessionToken) {
+    options.headers["X-Amz-Security-Token"] = options.sessionToken
+  }
 
   if (target) {
     options.headers["X-Amz-Target"] = target;
   }
 
+  var svc = service.split(".").pop()
   var credential = options.key +
     "/" +
-    createCredentialScope(options.timestamp, region, service);
+    createCredentialScope(options.timestamp, options.region, svc);
   var signedHeaders = createSignedHeaders(options.headers);
   var signature = generateSignature(
-    service, method, path, query, body,
-    options.headers, options.doubleEscape, options.timestamp, region, options.secret);
+    svc, method, path, query, body,
+    options.headers, options.doubleEscape, options.timestamp, options.region, options.secret);
 
   var signatureHeader = "AWS4-HMAC-SHA256 Credential=".concat(
     credential,
-    " SignedHeaders=",
+    ", SignedHeaders=",
     signedHeaders,
-    " Signature=",
+    ", Signature=",
```

Comment:
From https://docs.aws.amazon.com/general/latest/gr/sigv4-add-signature-to-request.html#sigv4-add-signature-auth-header

>There is no comma between the algorithm and Credential. However, the SignedHeaders and Signature are separated from the preceding values with a comma.